File: 2 лаба/E2.cpp
#include <iostream>
#include <fstream>
#include <vector>
using namespace std;

int kth(vector<int> &mas, int l, int r, int k) {

    int j = r, i = l;
    int s = (l+r)/2;
    int x = mas[s];
    if (l==r) {
        return mas[k];
    }
    while (i<=j) {
        while (mas[i]<x){
            i++;
        }
        while (mas[j]>x){
            j--;
        }
        if (i<=j) {
            swap(mas[i],mas[j]);
            i++;
            j--;
        }
    }
    if ((k<=j) && (l<=k)){
        return kth(mas,l,j,k);
    }
    if ((i<=k) && (k<=r)) {
        return kth(mas,i,r,k);

    }
}

int main(){

    ifstream fin("kth.in");
    ofstream fout("kth.out");

    int N, K, A, B, C, a1, a2;
    fin >> N >> K;
    fin >> A >> B >> C >> a1 >> a2;

    vector<int> nums(N);
    nums = {a1, a2};
    for (int i = 2; i < N; ++i) {
        nums[i] = (A * nums[i - 2] + B * nums[i - 1] + C);
    }

    kth(nums,0,N-1,K-1);
    int ans = nums[K-1];


    fout << ans ;
    fin.close();
    fout.close();

    return 0;
}

File: 2 lab - Merge Sort, Quick Sort/E2.py
import random


def quick_sort(nums, fst, lst):
   if fst >= lst: return
 
   i, j = fst, lst
   pivot = nums[random.randint(fst, lst)]
 
   while i <= j:
       while nums[i] < pivot: i += 1
       while nums[j] > pivot: j -= 1
       if i <= j:
           nums[i], nums[j] = nums[j], nums[i]
           i, j = i + 1, j - 1
   quick_sort(nums, fst, j)
   quick_sort(nums, i, lst)


fin = open("kth.in",  "r")
fout = open("kth.out", "w")

Array = []
Input_nums = list(map(int, fin.read().split()))
N, K, A, B, C = Input_nums[0], Input_nums[1], Input_nums[2], Input_nums[3], Input_nums[4]
Array.append(Input_nums[5])
Array.append(Input_nums[6])

for i in range(2, N):

	Next_num = bytearray(((A*Array[i-2] + B*Array[i-1] + C).to_bytes(32, byteorder='little', signed=True)))
	print(Next_num)
	Array.append(int.from_bytes(Next_num[0:5], byteorder='little', signed=True))


print(Array)






input()
fin.close()
fout.close()
